mypy/semanal.py

```diff
@@ -6462,6 +6471,34 @@ def set_future_import_flags(self, module_name: str) -> None:
     def is_future_flag_set(self, flag: str) -> bool:
         return self.modules[self.cur_mod_id].is_future_flag_set(flag)
 
+    def parse_dataclass_transform_spec(self, call: CallExpr) -> DataclassTransformSpec:
+        """Build a DataclassTransformSpec from the arguments passed to the given call to
+        typing.dataclass_transform."""
+        parameters = DataclassTransformSpec()
+        for name, value in zip(call.arg_names, call.args):
+            if name not in KNOWN_DATACLASS_TRANSFORM_PARAMETERS:
+                self.fail(f"unrecognized dataclass_transform parameter '{name}'", call)
+
+            # field_specifiers is currently the only non-boolean argument; check for it first so
+            # so the rest of the block can fail through to handling booleans
+            if name == "field_specifiers":
+                self.fail("field_specifiers support is currently unimplemented", call)
+                continue
+
+            boolean = self.parse_bool(value)
+            if boolean is None:
+                self.fail(f"{name} argument must be True or False.", call)
+                continue
+
+            if name == "eq_default":
+                parameters.eq_default = boolean
+            elif name == "order_default":
+                parameters.order_default = boolean
+            elif name == "kw_only_default":
+                parameters.kw_only_default = boolean
```

Comment:
Maybe `assert_never(name)` so we can be sure we covered all cases.

Reply:
moved the `self.fail` down here, which should cover both cases